_topics/en/op_checksigfromstack.md

```diff
@@ -0,0 +1,179 @@
+    accept its unconfirmed transactions as reliable payments.  To
+    demonstrate its good faith, the service can use OP_CSFS to offer
+    payment of a bond to any user that can prove the same key was used
+    to create two different signatures for transactions spending the
+    same UTXO.  <!-- Source: "These new opcodes have several use cases,
+    including double-spent protection bonds"
+    https://web.archive.org/web/20160828061959/http://elementsproject.org/elements/opcodes
+    -->
+
+      This use of `OP_CSFS` can be compared to [single-show
+      signatures][] that allow anyone who sees two signatures from the
+      same key to derive the private key used to create them, allowing
+      them to spend any other funds secured by that key.
+
+  - **Transaction introspection:** If the same pubkey and signature pair
+    are valid both with `OP_CSFS` and `OP_CHECKSIG`, then the contents
+    of the arbitrary message passed to `OP_CSFS` is identical to the
+    serialized spending transaction (and other data) implicitly used
+    with `OP_CHECKSIG`.  This makes it possible to put a validated copy
+    of the spending transaction on the script evaluation stack where
+    other opcodes can run tests on it in order to enforce restrictions
+    on the spending transaction.
+
+      For example, if `OP_CSFS` had been available in 2015 and 2016, it
+      would've been possible to implement the features of [BIP65][]
+      `OP_CHECKLOCKTIMEVERIFY` (CLTV) and [BIP112][]
+      `OP_CHECKSEQUENCEVERIFY` (CSV) using without any consensus changes
+      just by writing a verification script.
+
+      Looking forward, `OP_CSFS` could also allow scripts to [implement
+      the features][oconnor generic] of the proposed [SIGHASH_NOINPUT][topic
+      sighash_noinput] and `SIGHASH_ANYPREVOUT` signature hashes, as
+      well as other opcode proposals such as
+      OP_CHECKTEMPLATEVERIFY.
+      Additionally, `OP_CSFS` would allow the creation of
+      [covenants][topic covenants] that restrict the way in which a set
+      of bitcoins may be spent---for example, a vault may
+      restrict its spending transaction to a small set of acceptable
+      scriptPubKeys to limit the risk of theft.
+
+      The strength of `OP_CSFS` is that it provides full introspection
+      of the signing transaction in a completely generic way.  Its
+      weakness is that it requires essentially adding a complete copy of
+      the signing transaction to the stack, which may significantly
+      increase the size of transactions that want to use `OP_CSFS` for
+      introspection.  By comparison, single-purpose introspection
+      opcodes such as CLTV and CSV use minimal overhead, but adding each
+      new special introspection opcode requires a consensus change and
+      it may not be possible to disable their use (even if they become
+      unpopular) without risking someone losing money.
+
+  ### Relationship to OP_CAT
```

Comment:
nit: this may look better as h4, for more obvious contrast with the h2 "Primary code and documentation" element below.

Reply:
I made the h3 styling bolder to hopefully distinguish it from h4, that way we don't have to implement your *craaaaaazy* idea of skipping a level.  Let me know if this is acceptable.

- Old: 
![2020-06-01-14_37_42_923289125](https://user-images.githubusercontent.com/61096/83442665-883a9d80-a416-11ea-958f-b7cc89777675.png)

- New:
![2020-06-01-14_37_19_090513960](https://user-images.githubusercontent.com/61096/83442680-8ec91500-a416-11ea-8d80-9ad516bfa056.png)

Reply:
Let's revert this. My suggestion to use h4 was to make it more obvious that it was a subsection of the summary, and not a sibling of the "Primary code..." and "Optech newsletter..." headings. To my eye, the boldening might make it look like it's a parent of those elements.

(I have ~zero talent for design, so feel free to tell me I'm talking nonsense).

This also has the undesirable side-effect of making all h3s on the site bold, including the newsletter links on our home page and newsletter index.